
uninline a buffer.c function


 fs/buffer.c |    2 	1 +	1 -	0 !
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rc9/fs/buffer.c
===================================================================
--- linux-2.6.23-rc9.orig/fs/buffer.c	2007-10-08 23:51:19.000000000 +0200
+++ linux-2.6.23-rc9/fs/buffer.c	2007-10-09 00:30:41.000000000 +0200
@@ -564,7 +564,7 @@
 /*
  * The buffer's backing address_space's private_lock must be held
  */
-static inline void __remove_assoc_queue(struct buffer_head *bh)
+static void __remove_assoc_queue(struct buffer_head *bh)
 {
 	list_del_init(&bh->b_assoc_buffers);
 	WARN_ON(!bh->b_assoc_map);
