---
 arch/i386/lib/usercopy.c |    2 	1 +	1 -	0 !
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.23-rc9/arch/i386/lib/usercopy.c
===================================================================
--- linux-2.6.23-rc9.orig/arch/i386/lib/usercopy.c	2007-07-09 01:32:17.000000000 +0200
+++ linux-2.6.23-rc9/arch/i386/lib/usercopy.c	2007-10-09 00:29:05.000000000 +0200
@@ -16,7 +16,7 @@
 
 static inline int __movsl_is_ok(unsigned long a1, unsigned long a2, unsigned long n)
 {
-#ifdef CONFIG_X86_INTEL_USERCOPY
+#if defined(CONFIG_CPU_SUP_INTEL) && defined(CONFIG_X86_INTEL_USERCOPY)
 	if (n >= 64 && ((a1 ^ a2) & movsl_mask.mask))
 		return 0;
 #endif
